# Práctica 1.1 — Probabilidad y Estadística
## Análisis de la evolución poblacional mundial


## Introducción

El **objetivo** de esta práctica es identificar correctamente la fuente de un conjunto de datos real, cargarlo y explorarlo con Python, y calcular sobre él las principales medidas de estadística descriptiva.

El **dataset** utilizado es *Annual change in population* de **Our World in Data**, construido a partir de *World Population Prospects (2024)* de las Naciones Unidas.

La **variable principal** es `Annual change in population`, es decir, el cambio neto anual de la población, calculado como la diferencia entre la población al 1 de julio de dos años consecutivos.

Lo que se pretende analizar es **cómo cambia la población de las distintas entidades a lo largo del tiempo**: cuántas entidades y años cubre el dataset, qué valores faltantes tiene y qué magnitud y dispersión presenta el cambio poblacional, tanto a nivel global como en un subconjunto concreto (Granada, 1983–2021).

El análisis se realiza con **Python**, utilizando **pandas**, **NumPy** y **Jupyter Notebook**.


### Actividad 1 — Identificación de la fuente

Reconocer la procedencia del conjunto de datos antes de utilizarlo y documentar correctamente su fuente.


**Documentación de la fuente**

Los datos utilizados en esta práctica provienen de **Our World in Data (OWID)**. La fuente original es **United Nations, World Population Prospects (2024)**, procesada por Our World in Data.

**CITA**

UN, World Population Prospects (2024) – processed by Our World in Data. “Annual change in population – UN WPP” [dataset]. United Nations, “World Population Prospects”; United Nations, “World Population Prospects - Interim Update” [original data].

**Preguntas**

**¿Quién produjo originalmente los datos?**  
Las Naciones Unidas (United Nations), a través de su publicación *World Population Prospects (2024)*.

**¿Qué organización los procesa para su utilización en Our World in Data?**  
Our World in Data (OWID), que estandariza los datos y calcula el indicador para su uso en sus visualizaciones.

**¿Qué representa la variable que analizarás?**  
La variable *Annual change in population* (cambio anual de la población) representa el **cambio neto anual de la población**, calculado como la **diferencia entre la población al 1 de julio de dos años consecutivos**. Al ser un cambio *neto*, refleja de forma combinada los **nacimientos, las muertes y la migración**. Su unidad son **personas** y cubre el rango 1951–2023 (el conjunto incluye además una columna con proyecciones hasta 2100 basadas en el escenario medio de la ONU).

In [1]:
import pandas as pd
import numpy as np

### Actividad 2 — Carga y exploración de datos

**2.2 Cargar el archivo CSV**

Se carga el archivo con `pd.read_csv()`. La ruta se adapta a la estructura real de la carpeta: la subcarpeta se llama `annual-population-growth` (con guiones).

In [2]:
import os

# La ruta se adapta segun desde donde se ejecute el notebook.
rutas_posibles = [
    "data/raw/annual-population-growth/annual-population-growth.csv",       # cwd = raiz del proyecto (BigData/)
    "../data/raw/annual-population-growth/annual-population-growth.csv",    # cwd = carpeta notebooks/
]
ruta_csv = next((p for p in rutas_posibles if os.path.exists(p)), rutas_posibles[0])

df = pd.read_csv(ruta_csv)
print("Archivo cargado desde:", ruta_csv)

Archivo cargado desde: ../data/raw/annual-population-growth/annual-population-growth.csv


**2.3 Explorar el DataFrame**

In [3]:
# Primeras filas
df.head(13200)

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN
...,...,...,...,...,...
13195,Grenada,GRD,2096,NaN,-822.0
13196,Grenada,GRD,2097,NaN,-816.0
13197,Grenada,GRD,2098,NaN,-807.0
13198,Grenada,GRD,2099,NaN,-792.0


In [4]:
# Dimensiones (filas, columnas)
df.shape

(38400, 5)

In [5]:
# Nombres de columnas
df.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [6]:
# Tipos de datos
df.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [7]:
# Estadisticas descriptivas
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [8]:
# Cantidad de valores faltantes por columna
df.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [9]:
# Cantidad de entidades diferentes
df["Entity"].nunique()

256

In [10]:
df_2020 = df[df["Year"]==2020]
df_2020.head(3)

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
69,Afghanistan,AFG,2020,1212852.0,NaN
219,Africa (UN),UN_AFR,2020,32815675.0,NaN
369,Albania,ALB,2020,-13061.0,NaN


In [11]:
df_2020_ext_col1= df_2020[["Entity","Year"]]
df_2020_ext_col1.head(3)
df_2020_ext_col2= df[df["Year"]== 2020][["Entity", "Year"]]
df_2020_ext_col2.shape

(256, 2)

In [12]:
df_2005_2020 = df[ (df["Year"] >= 2005) & (df["Year"] <= 2020) & (df["Entity"] == "Mexico") ]
df_2005_2020.shape

(16, 5)

In [13]:
df_2005_2020.head(3)

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22254,Mexico,MEX,2005,1417373.0,NaN
22255,Mexico,MEX,2006,1442162.0,NaN
22256,Mexico,MEX,2007,1520690.0,NaN


**Preguntas de la Actividad 2**

**¿Cuántas filas y columnas tiene el dataset?**  
Según `df.shape`, el dataset tiene **38 400 filas y 5 columnas**.

**¿Qué tipo de datos contiene la variable `Year`?**  
Según `df.dtypes`, la columna `Year` es de tipo **entero (`int64`)**.

**¿Existen valores faltantes?**  
Sí, sí existen valores faltantes. Según `df.isnull().sum()` se distribuyen así:

- `Code`: 1 200 valores faltantes.
- `Annual change in population`: 19 712 valores faltantes.
- `Annual population change (Projected)`: 18 688 valores faltantes.

Las columnas `Entity` y `Year` **no** tienen valores faltantes (0 en ambas). En total hay 39 600 valores faltantes en el dataset. Los faltantes en las dos columnas de datos son esperables: cada fila trae solo una de las dos series (estimaciones históricas 1951–2023 o proyecciones 2024–2100), y `Code` queda vacío para entidades agregadas (regiones) que no tienen código ISO.

**¿Cuántas entidades diferentes contiene el dataset?**  
Según `df["Entity"].nunique()`, el dataset contiene **256 entidades** diferentes.

### Interpretación

El dataset tiene **38 400 filas y 5 columnas** y cubre **256 entidades** (países, regiones y agregados como *World*) a lo largo de los años **1951 a 2100**.

Los **valores faltantes no son un error del archivo**: cada fila trae únicamente una de las dos series. `Annual change in population` corresponde a las estimaciones históricas (1951–2023) y `Annual population change (Projected)` a las proyecciones (2024–2100), por lo que cada columna queda vacía en los años que no le corresponden. Los 1 200 faltantes de `Code` corresponden a entidades agregadas (regiones, grupos de ingreso) que no tienen código ISO.

En `df.describe()` la **media del cambio anual (2 063 019 personas) es muy superior a la mediana (46 898)** y la desviación estándar es enorme (9 440 859). Esto indica una distribución **fuertemente sesgada a la derecha**: la mayoría de las entidades son países con cambios poblacionales pequeños, pero unos pocos agregados (como *World* o continentes) alcanzan valores de decenas de millones y arrastran la media hacia arriba. Por eso, para analizar un caso concreto conviene trabajar sobre un subconjunto filtrado, como se hace en la Actividad 3.


### Actividad 3 — Estadística descriptiva

Cálculo de las medidas de tendencia central y de dispersión sobre la variable `Annual change in population`, para el subconjunto de **Granada (Grenada) entre 1983 y 2021**.


In [14]:
df_ext_EY= df[ (df["Year"] >= 1983) & (df["Year"] <= 2021) & (df["Entity"] == "Grenada") ]
df_ext_EY.head(5)


,Entity,Code,Year,Annual change in population,Annual population change (Projected)
13082,Grenada,GRD,1983,564.0,NaN
13083,Grenada,GRD,1984,555.0,NaN
13084,Grenada,GRD,1985,606.0,NaN
13085,Grenada,GRD,1986,683.0,NaN
13086,Grenada,GRD,1987,730.0,NaN


In [15]:
df_ext_EY.mean(numeric_only=True)

Year                                    2002.000000
Annual change in population              571.717949
Annual population change (Projected)            NaN
dtype: float64

In [16]:
df_ext_EY.median(numeric_only=True)

Year                                    2002.0
Annual change in population              564.0
Annual population change (Projected)       NaN
dtype: float64

In [17]:
df_ext_EY.var(numeric_only=True)

Year                                      130.00000
Annual change in population             32624.62888
Annual population change (Projected)            NaN
dtype: float64

In [18]:
df_ext_EY.std(numeric_only=True)

Year                                     11.401754
Annual change in population             180.622891
Annual population change (Projected)           NaN
dtype: float64

In [19]:
df_ext_EY.min(numeric_only=True)

Year                                    1983.0
Annual change in population              255.0
Annual population change (Projected)       NaN
dtype: float64

In [20]:
df_ext_EY.max(numeric_only=True)

Year                                    2021.0
Annual change in population              840.0
Annual population change (Projected)       NaN
dtype: float64

In [21]:
df.min(numeric_only=True)

Year                                        1951.0
Annual change in population             -3315935.0
Annual population change (Projected)   -22910351.0
dtype: float64

In [22]:
df.max(numeric_only=True)

Year                                        2100.0
Annual change in population             93371370.0
Annual population change (Projected)    72067360.0
dtype: float64

### Interpretación

Para **Granada entre 1983 y 2021**, el cambio anual de población presenta los siguientes valores:

| Medida | Valor |
|--------|-------|
| Media | 571.72 personas por año |
| Mediana | 564.00 personas por año |
| Varianza | 32 624.63 |
| Desviación estándar | 180.62 personas |
| Mínimo | 255 personas |
| Máximo | 840 personas |

La **media (571.72)** indica que, en promedio, Granada ganó alrededor de 572 habitantes por año en ese periodo. Como la **mediana (564)** es prácticamente igual a la media, la distribución es **aproximadamente simétrica**: no hay años atípicos que distorsionen el promedio.

La **desviación estándar (180.62)** mide cuánto se separan los años individuales de ese promedio; comparada con una media de 571.72 es una dispersión moderada, lo que refleja un crecimiento sostenido pero irregular. La **varianza (32 624.63)** es esa misma dispersión al cuadrado, por lo que no se expresa en personas y se usa sobre todo como paso intermedio.

El **mínimo (255)** y el **máximo (840)** muestran que en todos los años del periodo el cambio fue **positivo**: Granada nunca perdió población entre 1983 y 2021, solo creció más rápido en unos años que en otros.

Al comparar con el dataset completo (`df.min()` y `df.max()`), el rango global va de **-3 315 935** hasta **93 371 370** personas: los valores negativos corresponden a entidades que perdieron población y los máximos a agregados mundiales. Esto confirma la enorme diferencia de escala entre un país pequeño como Granada y los agregados.


### Actividad 4 — Comprobación del entorno

Verificación de las versiones de Python y de las librerías utilizadas en la práctica.


In [23]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)


Python: 3.14.7 (v3.14.7:823f0323ee6, Aug  5 2026, 07:07:01) [Clang 21.0.0 (clang-2100.1.1.101)]
Pandas: 3.0.5
NumPy: 2.5.2


### Interpretación

Esta celda documenta el entorno exacto con el que se ejecutó el notebook. Sirve para que el análisis sea **reproducible**: si otra persona obtiene resultados distintos, lo primero que debe compararse son estas versiones frente a las declaradas en `requirements.txt`.
